Lost use case with `no-reserved-keys` to `quote-props` merge

Previously I had `quote-props` turned off and wanted errors for `no-reserved-keys` but that now seems impossible. In other words, I don't care about property quoting, except if it's a keyword.

Another way to look at it is wanting `as-needed` to have an option to ignore unnecessarily quoted properties.

I'm using v1.1.0. For the above code I get

```
  3:5  error  Unnecessarily quoted property `unnecessary` found  quote-props
  4:5  error  Unquoted reserved word `if` used as key            quote-props
```

Which is correct for the current version, I'm just saying that I've lost the pre-1.0 behaviour where I could get the second error only.

For example, with v0.24.1 I would have used the rules `/*eslint quote-props: 0, no-reserved-keys: 2 */` to get the output:

```
  2:10  error  Reserved word 'if' used as key  no-reserved-keys
```

--- a/docs/rules/quote-props.md
+++ b/docs/rules/quote-props.md
@@ -160,6 +160,23 @@ var x = {
 };
 ```
 
+Another modifier for this rule is the `unnecessary` option which defaults to `true`. Setting this to `false` will prevent the rule from complaining about unnecessarily quoted properties. This comes in handy when you _only_ care about quoting keywords.
+
+```json
+{
+    "quote-props": [2, "as-needed", {"keywords": true, "unnecessary": false}]
+}
+```
+
+When `unnecessary` is set to `false`, the following patterns _stop_ becoming warnings:
+
+```
+var x = {
+    "while": 1,
+    "foo": "bar"  // Would normally have caused a warning
+};
+```
+
 #### consistent
 
 When configured with `"consistent"`, the patterns below are considered warnings. Basically `"consistent"` means all or no properties are expected to be quoted, in other words quoting style can't be mixed within an object. Please note the latter situation (no quotation at all) isn't always possible as some property names require quoting.

--- a/lib/rules/quote-props.js
+++ b/lib/rules/quote-props.js
@@ -21,6 +21,7 @@ module.exports = function(context) {
 
     var MODE = context.options[0],
         KEYWORDS = context.options[1] && context.options[1].keywords,
+        CHECK_UNNECESSARY = !context.options[1] || context.options[1].unnecessary !== false,
 
         MESSAGE_UNNECESSARY = "Unnecessarily quoted property `{{property}}` found.",
         MESSAGE_UNQUOTED = "Unquoted property `{{property}}` found.",
@@ -75,7 +76,7 @@ module.exports = function(context) {
                 return;
             }
 
-            if (areQuotesRedundant(tokens)) {
+            if (CHECK_UNNECESSARY && areQuotesRedundant(tokens)) {
                 context.report(node, MESSAGE_UNNECESSARY, {property: key.value});
             }
         } else if (KEYWORDS && key.type === "Identifier" && isKeyword(key.name)) {
@@ -191,6 +192,9 @@ module.exports.schema = {
                     "type": "object",
                     "properties": {
                         "keywords": {
+                            "type": "boolean"
+                        },
+                        "unnecessary": {
                             "type": "boolean"
                         }
                     },

--- a/tests/lib/rules/quote-props.js
+++ b/tests/lib/rules/quote-props.js
@@ -48,7 +48,8 @@ ruleTester.run("quote-props", rule, {
         { code: "({ '@': 0, 'B': 0 })", options: ["consistent-as-needed"] },
         { code: "({ a: 0, 'if': 0 })", options: ["as-needed", {keywords: true}] },
         { code: "({ a: 0, 'while': 0 })", options: ["as-needed", {keywords: true}] },
-        { code: "({ a: 0, 'volatile': 0 })", options: ["as-needed", {keywords: true}] }
+        { code: "({ a: 0, 'volatile': 0 })", options: ["as-needed", {keywords: true}] },
+        { code: "({'unnecessary': 1, 'if': 0})", options: ["as-needed", {keywords: true, unnecessary: false}] }
     ],
     invalid: [{
         code: "({ a: 0 })",
@@ -131,6 +132,12 @@ ruleTester.run("quote-props", rule, {
         options: ["as-needed", {keywords: true}],
         errors: [{
             message: "Unquoted reserved word `while` used as key.", type: "Property"
+        }]
+    }, {
+        code: "({'unnecessary': 1, if: 0})",
+        options: ["as-needed", {keywords: true, unnecessary: false}],
+        errors: [{
+            message: "Unquoted reserved word `if` used as key.", type: "Property"
         }]
     }]
 });

